src/app/settings/action-triggers/components/action-triggers-list/action-triggers-list.component.ts

```diff
@@ -0,0 +1,179 @@
+  }
+
+  private deleteActionTriggerById(actionTriggerId: string, deleteAllSelected?: boolean): void {
+    if (deleteAllSelected) {
+      for (let deleteActionTrigger of this.selectedActionTriggers) {
+        this.actionTriggerService.deleteActionTrigger(deleteActionTrigger.id).subscribe()
+      }
+    } else {
+      this.actionTriggerService.deleteActionTrigger(actionTriggerId).subscribe()
+    }
+  }
+
+  public actionTriggerCopy(actionTrigger: ActionTriggerWithActionInfo<KeyboardAndSelectionTriggerData>): void {
+    const copyPayload: CreateActionTrigger<KeyboardTriggerData> = {
+      actionId: actionTrigger.actionId,
+      data: { keys: actionTrigger.data.keys, label: actionTrigger.data.label, actionArguments: actionTrigger.data.actionArguments as number },
+    }
+    this.actionTriggerService.createActionTrigger(copyPayload).subscribe()
+  }
+
+  public selectNewSort(sortOption: string): void {
+    this.newSortSelect.emit(sortOption as ActionTriggerSortKeys)
+  }
+
+  public actionWithSelected(userAction: string): void {
+    if (userAction === UserActionsWithSelectedTriggers.DELETE) {
+      this.dialogService.createConfirmDialog($localize`action-triggers.delete-selected.label`, $localize`action-triggers.delete-selected.confirmation`, 'Delete', () =>
+        this.deleteActionTriggerById(this.selectedActionTriggers[0].id, true)
+      )
+    } else if (userAction === UserActionsWithSelectedTriggers.EXPORT) {
+      this.exportSelectedTriggers()
+    } else if (userAction === UserActionsWithSelectedTriggers.TOGGLE_SELECT) {
+      this.toggleSelectUnselectAll()
+    } else if (userAction === UserActionsWithSelectedTriggers.DISABLE_SELECTION) {
+      this.selectMode = false
+      this.unselectAllActionsTriggers()
+      this.checkSelectedCount()
+    }
+  }
+
+  private get selectedActionTriggers(): ActionTriggerWithActionInfo<KeyboardAndSelectionTriggerData>[] {
+    return this.actionsTriggersList.filter(item => item.data.selected)
+  }
+
+  private checkSelectedCount(): void {
+    this.selectedCount = this.actionsTriggersList.filter(item => item.data.selected).length
+    const toggleSelectIndex = this.selectedTriggersOptions.findIndex(item => item.key === UserActionsWithSelectedTriggers.TOGGLE_SELECT)
+    this.selectedTriggersOptions.forEach(option => {
+      if (option.key === UserActionsWithSelectedTriggers.TOGGLE_SELECT) {
+        if (this.selectedCount === this.actionsTriggersList.length) {
+          this.selectedTriggersOptions[toggleSelectIndex].label = $localize`global.unselect-all.label`
+        } else {
+          this.selectedTriggersOptions[toggleSelectIndex].label = $localize`global.select-all.label`
+        }
+      }
+      if (option.key === UserActionsWithSelectedTriggers.EXPORT || option.key === UserActionsWithSelectedTriggers.DELETE) {
+        if (this.selectedCount > 0) {
+          option.disabled = false
+        } else {
+          option.disabled = true
+        }
+      }
+    })
+  }
```

Comment:
From the method name I don't know what the method does. One question that comes to my mind is what does it check? And with a return type of void, how do I know if the check was successful?

By reading the method I assume that this updates the dropdown labels and their availability. Perhaps the method could be called something like `updateSelectedActionTriggersOptions`.

Another thing that bugs me a bit, is that the parent component needs to alter the data structure of the action trigger (`KeyboardAndSelectionTriggerData`) for it to be usable. Instead, this component could have a collection of the action ids for the selected actions. That way, there is no need for caching the `selectedCount`.
By doing this, the need for the `CopyUtil` class in this file and in the `ActionTriggers` component will be eliminated.

In the suggestion below, I have taken the liberty to split up the method and use a functional approach for updating the options.

```typescript
  private selectedActionTriggerIds: Set<string> = new Set()

  private updateSelectedActionTriggersOptions(): void {
    this.selectedActionTriggerOptions = this.selectedActionTriggerOptions.map(this.updateSelectedActionTriggersOption.bind(this))
  }

  private updateSelectedActionTriggersOption(option: SofieDropdownOption): SofieDropdownOption {
    switch (option.key) {
      case UserActionsWithSelectedTriggers.TOGGLE_SELECT:
        return {
          ...option,
          label: this.areAllActionTriggersSelected() ? $localize`global.unselect-all.label` : $localize`global.select-all.label`
        }
      case UserActionsWithSelectedTriggers.EXPORT:
      case UserActionsWithSelectedTriggers.DELETE:
        return {
          ...option,
          disabled: this.isNoActionTriggerSelected()
        }
      default:
        return option
    }
  }

  private areAllActionTriggersSelected(): boolean {
    return this.actionTriggers.length === this.selectedActionTriggerIds.size
  }

  private isNoActionTriggerSelected(): boolean {
    return this.selectedActionTriggerIds.size === 0
  }
```

Reply:
Take little bit more changes, but I think it's become better now. You can check it, Done.